### PR TITLE
Stacker-DB Contract & Tests Setup

### PR DESCRIPTION
**Pull request by setzeus**
## Summary of Changes
With DR/hackathon behind us the Clarity WG focus is on the following contracts: 

1. sBTC
2. sBTC-DB
3. PoX-4
4. Stacker-DB

There's been a lot of back-&-forth on where exactly these boot / storage contracts will live. This PR is meant to be the first flag in the ground for these new contracts in terms of setup. This PR places stacker-db.clar inside the src/chainstate/stacks/boot directory & it's corresponding test in the contrib/core-contract-tests/tests directory. 

It's very likely that contrib/core-contract-tests evolves since it seems outdated from a Clarinet perspective (nvm the upcoming Clarinet-SDK); hopefully, however, it won't require a restructure / shift from the setup implemented here. 

Specifically, this PR introduces a stacker-db.clar contract & corresponding typescript tests. 

## Testing
Includes five unit tests for now that can be checked using the soon-to-be-outdated "clarinet test" command from within the contrib/core-contract-tests directory

### Risks
Inadequate setup down the line to maintain contracts in the boot directory & Clarinet testing separated in the contrib directory. Not sure how this will evolve when adding the remaining contracts and/or when updating Clarinet.

### How were these changes tested?
Using Clarinet built-in Deno testing setup found in the contrib directory.

## Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my code
- [X] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by friedger — COMMENTED** [review]
clarinet sdk has been release as version 1.+. Why using the deprecated method `clarinet test`?

**Comment by friedger** [comment]
The PR should go to branch `next` I think.

File: src/chainstate/stacks/boot/stacker-db.clar [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	(+ (get first-burnchain-block-height pox-info) (* cycle (get reward-cycle-length pox-info))))
+
+(define-read-only (burn-height-to-reward-cycle (height uint))
+	(/ (- height (get first-burnchain-block-height pox-info)) (get reward-cycle-length pox-info)))
+
+(define-read-only (current-pox-reward-cycle)
+	(burn-height-to-reward-cycle burn-block-height))
+
+;; Read
+ (define-read-only (stackerdb-get-config)
```
[the remark this answers is not in the document]
Comment (saralab) [reply src/chainstate/stacks/boot/stacker-db.clar:26]: +1 this

File: src/chainstate/stacks/boot/stacker-db.clar [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                max-writes: u4096,
+                max-neighbors: u32,
+                hint-replicas: (list )
+            }))
+
+(define-read-only (get-current-signer-slots)
+    (map-get? signer-slots-by-reward-cycle (current-pox-reward-cycle))
+)
+
+(define-read-only (get-any-signer-slots (cycle uint))
```
[the remark this answers is not in the document]
Comment (saralab) [reply src/chainstate/stacks/boot/stacker-db.clar:39]: I vote for get-signer-slots-by-cycle

**Pull request closed without merging**